pkg/server/setup.go

```diff
@@ -93,6 +94,10 @@ func setupCli(c *gin.Context) error {
 		return err
 	}
 
+	if err := setupProvider(&setup); err != nil {
```

Comment:
I'm pretty sure reordering these will break the ability to clone the git repo (since the directory will already exist)

Reply:
Correct me if I am wrong, but it looks like the `setupProvider` step does not directly touch the `~/workspace` directory.
For the GCP it writes the credentials to the `~/gcp.json` and for AWS it uses `aws configure set ...` command to alter config files in the `~/.aws` directory. It should not have an impact on the `setupGit` step and `~/workspace` directory creation during the `git clone`.

Reply:
Oh you're right, this makes sense